mesonbuild/mesonlib.py

```diff
@@ -877,7 +877,6 @@ def do_replacement(regex: T.Pattern[str], line: str, variable_format: str,
         start_tag = '${'
         backslash_tag = '\\${'
     else:
-        assert variable_format == 'meson'
```

Comment:
Why do you want to remove this assert?

Reply:
because it prevents the use of 'cmake@'  format.

Reply:
Then instead of removing the assert (which is there for a reason to catch bugs), you should fix it instead to also allow the `cmake@` format. Maybe something like `assert variable_format in ['meson', 'cmake@']`.

Reply:
:+1: